warehouse/ingest-core/src/test/resources/datawave/ingest/mapreduce/job/all-splits.txt

```diff
@@ -1,11 +1,11 @@
-shard	[B@57c8b24d
-shard	[B@39320a41
-shard	[B@43f4ebd
-shard	[B@584a5c78
-shard	[B@79fcf790
-shard1	[B@79fcf790
+shard	B0E=
```

Comment:
I'm probably just being overly paranoid. Sorry about that....

The thing that originally caught my attention was the fact that the data in these test files was changed, which seemed like a red flag, because, if we have to change existing data in order for it to be decoded successfully, then we could definitely have a problem, particularly if the data in question happens to already be on disk in Accumulo.

We had a similar situation a while back, where DW code was using Base64 encode/decode logic provided by a deprecated Infinispan dependency, and we found that we _did_ need to replace it with Java's mime encoder/decoder to ensure compatibility with existing data in Accumulo

That said, I just tried running some junits for these test files as they exist currently, and I apparently discovered the same thing you did....

I changed the decode logic in `NonShardedSplitsFile.Reader` to attempt both `java.util.Base64.getDecoder().decode` and `java.util.Base64.getMimeDecoder().decode` only to find that BOTH of those decoders fail with 'java.lang.IllegalArgumentException` on these files.

So, taking a closer look at the encoded values in these files, my new question is: Why the heck doesn't commons-codec's Base64 choke on  them as well? AFAIK, they don't appear to be valid Base64 according to _either_ RFC spec. So, still kinda worrisome that there's that much difference in the behavior between commons-codec and Java

However, I do concede that it would be problematic, in this case specifically, for us to use to the mime encoder to generate the split files, since we rely on the table name and the encoded  value being on a single line. Large-ish values getting split across lines would definitely break our file parsing.

And it does seem like all of data in question here (in terms of the entire PR) is probably all ephemeral config-related stuff. Not 100% sure about that though

Still deserves some thorough integration testing just to be sure :-)

Reply:
yeah I'm definitely doing a bunch of testing lol. the reason the test data got changed (the issue was originally found be a dev who isn't with us anymore so hopefully I'm saying the issue right) is because the data in the original test files were actually memory locations from something and not actual base64 so it was giving bad input to the encoders/decoders. 

From what I can find/see, the big difference between apache and java base64 is that if you give apache bad data it will just gobble it up and return bad data to you were as java wont take bad input and will properly throw an error. I'm running into a similar problem downstream where a group of unit tests had been feeding the decoder non-base64 and now that it wont accept bad input, it's failing the test.

this is some high quality pair-programming :)